packages/cloud_firestore/cloud_firestore/lib/src/query.dart

```diff
@@ -656,6 +656,7 @@ class _JsonQuery implements Query<Map<String, dynamic>> {
     bool hasIn = false;
     bool hasNotIn = false;
     bool hasNotEqualTo = false;
+    bool hasNotEqualToNotOperatorAndNotDocumentIdField = false;
```

Comment:
What a name haha

Reply:
ha, I think there's an extra `Not` in there 😄. It should be `hasNotEqualToOperatorAndNotDocumentIdField`. Still not great, but I need to capture this use case.